src/averageAges.js

```diff
@@ -9,17 +9,33 @@
  * To calculate century:
  * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
- * @param {object[]} people
- * @param {number} century - optional
  *
  * @return {number}
+ * @param year
  */
+
+const getCentury = year => Math.floor((year + 99) / 100);
+const getAge = (born, died) => died - born;
+const getAverageAge = (arr) => {
+  return arr.reduce((years, { born, died }) => years
+      + getAge(born, died), 0)
+    / arr.length;
+};
+const getGender = (gender) => {
+  return ({ sex }) => sex === gender;
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const allMen = people.filter(getGender('m'));
+  let menToCountAverageAge = [];
+
+  !century
+    ? menToCountAverageAge = allMen
+    : menToCountAverageAge = allMen.filter(({ died }) => {
+      return getCentury(died) === century;
+    });
```

Comment:
let's use if, it looks more readable
```suggestion
  if (century){
    allMen = allMen.filter(({ died }) => getCentury(died) === century);
 }
```

Reply:
в условии задачи было что иф низзя использовать